cmd/medius/images/push.go

```diff
@@ -25,67 +26,48 @@ import (
 func NewPublishImagesCommand(options *common.Options) *cobra.Command {
 	options.PublishImagesOptions = common.PublishImageOptions{
 		ForceBuild: false,
-		Workers:    1,
 	}
 
 	publishCmd := &cobra.Command{
 		Use:   "push",
 		Short: "Determine if containerdisks need an update and push an update to the target registry if needed",
 		Run: func(cmd *cobra.Command, args []string) {
-			errChan := make(chan error, options.PublishImagesOptions.Workers)
-			jobChan := make(chan api.Artifact, options.PublishImagesOptions.Workers)
+			results := map[string]*api.ArtifactResult{}
+
+			m := &sync.Mutex{}
+			err := spawnWorkers(cmd.Context(), options, func(a api.Artifact) error {
+				result, err := buildAndPublish(cmd.Context(), a, options, time.Now())
+				if result != nil {
+					m.Lock()
+					results[a.Metadata().Describe()] = result
```

Comment:
I would have collected the results via  a channel to avoid the lock, but this works too.